src/default_allocator.cpp

```diff
@@ -34,7 +34,7 @@ DefaultAllocator::Deallocate(void* p) {
     if (!p) {
         return;
     }
-    std::lock_guard<std::mutex> guard(set_mutex_);
+    std::unique_lock<std::mutex> guard(set_mutex_);
```

Comment:
Why change to `unique_lock` ? here no need manual lock or unlock

Reply:
done